packages/dds/tree/src/simple-tree/api/verboseTree.ts

```diff
@@ -0,0 +1,409 @@
+ */
+export type VerboseTree<THandle = IFluidHandle> =
+	| VerboseTreeNode<THandle>
+	| Exclude<TreeLeafValue, IFluidHandle>
+	| THandle;
+
+/**
+ * Verbose encoding of a {@link TreeNode}.
+ * @remarks
+ * This is verbose meaning that every {@link TreeNode} has an explicit `type` property, and `fields`.
+ * This allowed VerboseTreeNode to be unambiguous regarding which type each node is without relying on symbols or hidden state.
+ *
+ * Any IFluidHandle values have been replaced by `THandle`. If the `THandle` is JSON compatible, then this type is JSON compatible as well.
+ *
+ * @privateRemarks
+ * This type is only used for data which is copied into and out of the tree.
+ * When being copied out, its fine to have the data be mutable since its a copy.
+ *
+ * When being copied in, we don't need to mutate, so we could use a readonly variant of this type.
+ * however the copy in case (createFromVerbose) probably isn't harmed much by just reusing this type as is,
+ * since if the caller has immutable data, TypeScript doesn't prevent assigning immutable data to a mutable type anyway.
+ * Also relaxing the input methods to take readonly data would be a non-breaking change so it can be done later if desired.
+ *
+ * This format is simple-tree specialized alternative to {@link JsonableTree}.
+ * This format allows for all simple-tree compatible trees to be represented.
+ *
+ * Unlike `JsonableTree`, leaf nodes are not boxed into node objects, and instead have their schema inferred from the value.
+ * Additionally, sequence fields can only occur on a node that has a single sequence field (with the empty key) replicating the behavior of simple-tree ArrayNodes.
+ */
+export interface VerboseTreeNode<THandle = IFluidHandle> {
+	/**
+	 * The meaning of this node.
+	 * Provides contexts/semantics for this node and its content.
+	 * Typically used to associate a node with metadata (including a schema) and source code (types, behaviors, etc).
+	 */
+	type: string;
+
+	/**
+	 * Content of this node.
+	 * For array nodes, an array of children.
+	 * For map and object nodes, an object which children under keys.
+	 * @remarks
+	 * For object nodes, the keys could be either the stored keys, or the property keys depending on usage.
+	 */
+	fields:
+		| VerboseTree<THandle>[]
+		| {
+				[key: string]: VerboseTree<THandle>;
+		  };
+}
+
+/**
+ * Options for how to interpret a `VerboseTree<TCustom>` when schema information is available.
+ */
+export interface ParseOptions<TCustom> {
+	/**
+	 * Fixup custom input formats.
+	 * @remarks
+	 * Main usage is translate some JSON compatible handle format into actual IFluidHandles.
+	 */
+	valueConverter(data: VerboseTree<TCustom>): TreeLeafValue | VerboseTreeNode<TCustom>;
+	/**
+	 * If true, interpret the input keys of object nodes as stored keys.
+	 * If false, interpret them as property keys.
```

Comment:
I think in other places we refer to these as "view" keys. Not sure which terminology is used more, but it would be good for us to settle on a single term.

Reply:
Do you have an example of such a place? FieldProps,key's docs calls them "property keys". I think that aligns with our API which uses them as the keys for properties on object nodes (ex: Object.prototype.hasOwnProperty calles them properties, and TypeScript types the parameter "key" parameter  as "PropertyKey")

Reply:
Note that the term "view schema" is generally not used in our public APi as its the only public schema system we have. Nearly everything thats public is implicitly "view".

Reply:
Yeah, a brief search didn't reveal the "view key" terminology in public APIs. I was thinking it ended up getting surfaced through `FieldProps` / `FieldSchema` / `treeNodeApi`, but that doesn't seem to be the case.

Still, we may want to do an audit of our non-public docs to unify that terminology. Do we have a glossary of terms anywhere?